lib/match/runner.rb

```diff
@@ -90,5 +96,28 @@ def profile(params: nil, certificate_id: nil)
 
       return uuid
     end
+
+    def device_count_different?(params: nil, profile: nil)
+      if profile && params[:force_for_new_devices]
+        parsed = FastlaneCore::ProvisioningProfile.parse(profile)
+        uuid = parsed["UUID"]
+        type = params[:type]
+
+        if type == "appstore"
+          profiles = Spaceship.provisioning_profile.app_store
+        elsif type == "adhoc"
+          profiles = Spaceship.provisioning_profile.ad_hoc
+        else
+          profiles = Spaceship.provisioning_profile.development
+        end
+
+        app_profiles = profiles.find_by_bundle_id(params[:app_identifier])
+        portal_profile = app_profiles.detect { |i| i.uuid == uuid }
```

Comment:
If we already have the profile UUID we should be able to find it directly by just iterating over
```ruby
Spaceship::ProvisioningProfile.all.detect { |i| i.uuid == uuid }
```
This way it also works for enterprise profiles and other types in the future